call-home/src/bin/callhome/main.rs

```diff
@@ -256,5 +294,81 @@ async fn generate_report(
     report.storage_media = StorageMedia::new(valid_disks);
 
     report.nexus = Nexus::new(event_data);
+
+    let logs = logs.lock().unwrap();
+    for log in logs.iter() {
+        let log_string = format!(
+            "[{} {} {}:{}] {}",
+            log.timestamp, log.level, log.file, log.line, log.message
+        );
+        report.logs.push(log_string);
+    }
     report
 }
+
+// Define the LogsLayer
+struct LogsLayer {
+    logs: Arc<Mutex<Vec<LogEntry>>>,
+}
+
+impl LogsLayer {
+    fn new(logs: Arc<Mutex<Vec<LogEntry>>>) -> Self {
+        LogsLayer { logs }
+    }
+}
+
+impl<S> Layer<S> for LogsLayer
+where
+    S: Subscriber,
+{
+    fn on_event(&self, event: &Event, _ctx: Context<S>) {
+        let timestamp = chrono::Utc::now().to_rfc3339();
+        let level = event.metadata().level();
+        let file = event.metadata().file().unwrap_or("").to_string();
+        let line = event.metadata().line().unwrap_or(0);
+
+        let mut visitor = LogVisitor::new();
+        event.record(&mut visitor);
+
+        let log_entry = LogEntry {
+            timestamp,
+            level: level.to_string(),
+            file,
+            line,
+            message: visitor.log,
+        };
+
+        // Only capture logs of level ERROR
+        if level == &Level::ERROR {
```

Comment:
I wonder if we also only want to capture logs with a certain field attribute. Example let's say one of the libraries starts generating a lot of error logs, this would generate a report with lots of data.